sdks/java/io/solace/src/test/java/org/apache/beam/sdk/io/solace/data/SolaceTest.java

```diff
@@ -0,0 +1,146 @@
+  String attachmentString = "some attachment";
+  ByteString attachment = ByteString.copyFrom(attachmentString, StandardCharsets.UTF_8);
+
+  @Test
+  public void testRecordEquality() {
+    Solace.Record obj1 =
+        Solace.Record.builder()
+            .setDestination(destination)
+            .setMessageId(messageId)
+            .setExpiration(expiration)
+            .setPriority(priority)
+            .setRedelivered(redelivered)
+            .setReplyTo(replyTo)
+            .setReceiveTimestamp(receiveTimestamp)
+            .setSenderTimestamp(senderTimestamp)
+            .setSequenceNumber(sequenceNumber)
+            .setTimeToLive(timeToLive)
+            .setPayload(payload)
+            .setAttachmentBytes(attachment)
+            .build();
+
+    Solace.Record obj2 =
+        Solace.Record.builder()
+            .setDestination(destination)
+            .setMessageId(messageId)
+            .setExpiration(expiration)
+            .setPriority(priority)
+            .setRedelivered(redelivered)
+            .setReplyTo(replyTo)
+            .setReceiveTimestamp(receiveTimestamp)
+            .setSenderTimestamp(senderTimestamp)
+            .setSequenceNumber(sequenceNumber)
+            .setTimeToLive(timeToLive)
+            .setPayload(payload)
+            .setAttachmentBytes(attachment)
+            .build();
+
+    Solace.Record obj3 =
+        Solace.Record.builder()
+            .setDestination(destination)
+            .setMessageId(messageId)
+            .setExpiration(expiration)
+            .setPriority(priority)
+            .setRedelivered(!redelivered)
+            .setReplyTo(replyTo)
+            .setReceiveTimestamp(receiveTimestamp)
+            .setSenderTimestamp(senderTimestamp)
+            .setSequenceNumber(sequenceNumber)
+            .setTimeToLive(timeToLive)
+            .setPayload(payload)
+            .setAttachmentBytes(attachment)
+            .build();
+
+    Assert.assertEquals(obj1, obj2);
+    Assert.assertNotEquals(obj1, obj3);
+    Assert.assertEquals(obj1.hashCode(), obj2.hashCode());
+    Assert.assertEquals(obj1.getDestination(), destination);
+    Assert.assertEquals(obj1.getMessageId(), messageId);
+    Assert.assertEquals(obj1.getExpiration(), expiration);
+    Assert.assertEquals(obj1.getPriority(), priority);
+    Assert.assertEquals(obj1.getRedelivered(), redelivered);
+    Assert.assertEquals(obj1.getReplyTo(), replyTo);
+    Assert.assertEquals(obj1.getReceiveTimestamp(), receiveTimestamp);
+    Assert.assertEquals(obj1.getSenderTimestamp(), senderTimestamp);
+    Assert.assertEquals(obj1.getSequenceNumber(), sequenceNumber);
+    Assert.assertEquals(obj1.getTimeToLive(), timeToLive);
+    Assert.assertEquals(obj1.getPayload().toString(StandardCharsets.UTF_8), payloadString);
+    Assert.assertEquals(
+        obj1.getAttachmentBytes().toString(StandardCharsets.UTF_8), attachmentString);
+  }
+
+  @Test
+  public void testRecordNullability() {
```

Comment:
Similar to the comment for testEquality, AutoValue will determine what fields can be null or not using the Nullable annotation. It may be that this test isn't needed either.